staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/meta.go

```diff
@@ -148,6 +150,89 @@ func GetTagStructTypeOrDie(dataStruct interface{}) reflect.Type {
 	return t
 }
 
+type PatchMetaFromOpenAPIV3 struct {
+	// SchemaList is required to resolve OpenAPI V3 references
+	SchemaList map[string]*spec.Schema
+	Schema     *spec.Schema
+}
+
+func (s PatchMetaFromOpenAPIV3) traverse(key string) (PatchMetaFromOpenAPIV3, error) {
+	if s.Schema == nil {
+		return PatchMetaFromOpenAPIV3{}, nil
+	}
+	if len(s.Schema.Properties) == 0 {
+		return PatchMetaFromOpenAPIV3{}, fmt.Errorf("unable to find api field \"%s\"", key)
```

Comment:
Should you look at additionalProperties?

Reply:
Does it make a difference to look at additionalProperties? It's only necessary if we have a complex map and need to traverse further to look for merge/patch options. k/k generally uses association lists for more complex types which do not touch additionalProperties. k/k has simple maps (eg: `map[string]string`) which do not need to recurse to the type of the map.

Reply:
If we don't use additionalProperties in k8s then yeah, it's fine, but I don't know if that's 100% true?

Reply:
We use additionalProperties for simple maps in k/k, mainly `map[string]string` which is covered by test cases involving SimpleMap (https://github.com/kubernetes/kubernetes/blob/31140a5f97f89495ddfd748dc4d283832da672eb/staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go#L108)

Reply:
Yeah, I couldn't find any in k8s.io/api, we're probably good, thanks.